src/components/top-bar/top-bar.tsx

```diff
@@ -0,0 +1,28 @@
+import type { FC, HTMLAttributes } from 'react'
+import {
+  ElTopBarMainNav,
+  ElTopBarSearchContainer,
+  ElTopBar,
+  ElTopBarMenuButtonContainer,
+  ElTopBarSecondaryNav,
+} from './styles'
+
+const TopBar: FC<HTMLAttributes<HTMLDivElement>> & {
+  MainNavigations: typeof ElTopBarMainNav
+  SecondaryNavigations: typeof ElTopBarSecondaryNav
+  SearchContainer: typeof ElTopBarSearchContainer
+  MenuButtonContainer: typeof ElTopBarMenuButtonContainer
+} = ({ children, ...props }) => {
+  return (
+    <nav>
+      <ElTopBar {...props}>{children}</ElTopBar>
+    </nav>
+  )
+}
+
+TopBar.MainNavigations = ElTopBarMainNav
```

Comment:
**issue:** can we please keep the name of these compound components matching the styled components? 🙏 

If you adopt the name changes I've suggested earlier, this would become:
```
TopBar.MainNav = ElTopBarMainNav
TopBar.SecondaryNav = ElTopBarSecondaryNav
TopBar.Search = ElTopBarSearch
TopBar.MobileNav = ElTopBarMobileNav
```

Reply:
yes, thanks for the example 👍